lib/mongo/collection/view/iterable.rb

```diff
@@ -70,7 +70,7 @@ def each
             # If a query with a limit is performed, the query cache will
             # re-use results from an earlier query with the same or larger
             # limit, and then impose the lower limit during iteration.
-            limit_for_cached_query = respond_to?(:limit) ? limit : nil
+            limit_for_cached_query = respond_to?(:limit) ? QueryCache.normalized_limit(limit) : nil
           end
 
           if block_given?
```

Comment:
Can't comment on line 89 because it's not in the diff but I'll ask on this one..

If we call `each` without a block, it'll just return an `Enumerable`, which won't have the cached limit applied. 

This case probably should be handled by always doing the positive case, and then calling `to_enum` on the result of it

Reply:
line 79 could end up being something like 
```ruby
result = if use_query_cache?
  @cursor.limited_to(limit) # is there a case where limit doesn't exist? CachingCursor inherits Cursor which implements limit, so don't see why we have the `respond_to?(:limit)` check
else
  @cursor
end

if block_given? 
  # iterate result and yield 
else
  result.to_enum
end
```

Just ideas at them moment, let me know what you think

Reply:
cc @p-mongo (not sure if you'd normally get notifications for these comments on someone elses PR)

Reply:
I do @mikebaldry , but thank you for the highlight. We'll discuss this PR with the team to figure out the best course of action.